File: go/difference-of-squares/difference_of_squares.go
package diffsquares

// Difference subtractrs SumOfSqureres(num) from SquareOfSum(num)
func Difference(num int) int {
	return SquareOfSum(num) - SumOfSquares(num)
}

// SquareOfSum returns the square of sum(1...Num) inclusive
func SquareOfSum(num int) int {
	summed := 0
	for i := 0; i <= num; i++ {
		summed += i
	}
	return summed * summed
}

// SumOfSquares returns the sum of squares from 1...Num inclusive
func SumOfSquares(num int) int {
	summedSquares := 0
	for i := 0; i <= num; i++ {
		summedSquares += i * i
	}
	return summedSquares
}

File: python/difference-of-squares/difference_of_squares.py
# exercism.io difference_of_squares

def square_of_sum(num):
	return sum(x for x in range(1, num+1)) ** 2

def sum_of_squares(num):
	return sum([x**2 for x in range(1, num+1)])

def difference(num):
	return square_of_sum(num) - sum_of_squares(num)
